src/main/java/ch/njol/skript/SkriptEventHandler.java

```diff
@@ -251,35 +249,40 @@ public static void registerBukkitEvent(Trigger trigger, Class<? extends Event> e
 	 * @param trigger The Trigger to unregister events for.
 	 */
 	public static void unregisterBukkitEvents(Trigger trigger) {
-		triggers.removeIf(pair -> {
-			if (pair.getSecond() != trigger)
-				return false;
+		Iterator<Map.Entry<Class<? extends Event>, Trigger>> entryIterator = triggers.entries().iterator();
+		Map.Entry<Class<? extends Event>, Trigger> entry;
+		entryLoop: while (entryIterator.hasNext()) {
+			entry = entryIterator.next();
+			if (entry.getValue() != trigger)
+				continue;
+			Class<? extends Event> event = entry.getKey();
 
-			HandlerList handlerList = getHandlerList(pair.getFirst());
-			assert handlerList != null;
+			// Remove the trigger from the map
+			entryIterator.remove();
 
+			// check if we can unregister the listener
 			EventPriority priority = trigger.getEvent().getEventPriority();
-			if (triggers.stream().noneMatch(pair2 ->
-				trigger != pair2.getSecond() // Don't match the trigger we are unregistering
-				&& pair2.getFirst().isAssignableFrom(pair.getFirst()) // Basic similarity check
-				&& priority == pair2.getSecond().getEvent().getEventPriority() // Ensure same priority
-				&& handlerList == getHandlerList(pair2.getFirst()) // Ensure same handler list
-			)) { // We can attempt to unregister this listener
-				Skript skript = Skript.getInstance();
-				for (RegisteredListener registeredListener : handlerList.getRegisteredListeners()) {
-					Listener listener = registeredListener.getListener();
-					if (
-						registeredListener.getPlugin() == skript
-						&& listener instanceof PriorityListener
-						&& ((PriorityListener) listener).priority == priority
-					) {
-						handlerList.unregister(listener);
-					}
+			for (Trigger t : triggers.get(event)) {
+				if (t.getEvent().getEventPriority() == priority) {
+					continue entryLoop;
 				}
 			}
 
-			return true;
-		});
+			// We can attempt to unregister this listener
+			HandlerList handlerList = getHandlerList(event);
+			assert handlerList != null;
```

Comment:
Can be null here, needs to return here, as Skript is printing the exception to console, not actually throwing, which is why the method returns null.